File: email-api/src/main/java/club/hosppy/email/dto/EmailRequest.java
package club.hosppy.email.dto;

import club.hosppy.email.EmailTmpl;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EmailRequest {

    @NotBlank(message = "Please provide an email")
    private String to;

    @NotBlank(message = "Please provide a subject")
    private String subject;

    private String name;

    @NotNull
    private EmailTmpl tmpl;

    private Map<String, Object> params;
}

File: src/main/kotlin/club/hosppy/email/dto/EmailRequest.kt
package club.hosppy.email.dto

import club.hosppy.email.EmailTmpl
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

data class EmailRequest(
    val to: @NotBlank(message = "Please provide an email") String,
    val subject: @NotBlank(message = "Please provide a subject") String,
    val name: String,
    val tmpl: @NotNull EmailTmpl,
    val params: Map<String, Any>? = null
)